infrastructure/src/main/java/org/corfudb/infrastructure/LocalMonitoringService.java

```diff
@@ -0,0 +1,138 @@
+    private static final Duration METRICS_POLL_INTERVAL = Duration.ofMillis(3000);
+
+    private final ScheduledExecutorService pollingService;
+
+    LocalMonitoringService(@NonNull ServerContext serverContext,
+                           @NonNull SingletonResource<CorfuRuntime> runtimeSingletonResource,
+                           @NonNull ClusterStateContext clusterStateContext) {
+        this.serverContext = serverContext;
+        this.runtimeSingletonResource = runtimeSingletonResource;
+        this.clusterStateContext = clusterStateContext;
+
+        this.pollingService = Executors.newSingleThreadScheduledExecutor(
+                new ThreadFactoryBuilder()
+                        .setDaemon(true)
+                        .setNameFormat(serverContext.getThreadPrefix() + "LocalMetricsPolling")
+                        .build());
+    }
+
+    private CorfuRuntime getCorfuRuntime() {
+        return runtimeSingletonResource.get();
+    }
+
+    /**
+     * Queries the local Sequencer Server for SequencerMetrics. This returns UNKNOWN if unable
+     * to fetch the status.
+     *
+     * @param layout Current Management layout.
+     * @return Sequencer Metrics.
+     */
+    private SequencerMetrics queryLocalSequencerMetrics(Layout layout) {
+        SequencerMetrics sequencerMetrics;
+        // This is an optimization. If this node is not the primary sequencer for the current
+        // layout, there is no reason to request metrics from this sequencer.
+        if (layout.getSequencers().get(0).equals(serverContext.getLocalEndpoint())) {
+            try {
+                sequencerMetrics = CFUtils.getUninterruptibly(
+                        getCorfuRuntime()
+                                .getLayoutView().getRuntimeLayout(layout)
+                                .getSequencerClient(serverContext.getLocalEndpoint())
+                                .requestMetrics());
+            } catch (ServerNotReadyException snre) {
+                sequencerMetrics = new SequencerMetrics(SequencerStatus.NOT_READY);
+            } catch (Exception e) {
+                log.error("Error while requesting metrics from the sequencer: ", e);
+                sequencerMetrics = new SequencerMetrics(SequencerStatus.UNKNOWN);
+            }
+        } else {
+            sequencerMetrics = new SequencerMetrics(SequencerStatus.UNKNOWN);
+        }
+        return sequencerMetrics;
+    }
+
+    /**
+     * Task to collect local node metrics.
+     * This task collects the following:
+     * Boolean Status of layout, sequencer and logunit servers.
+     * These metrics are then composed into ServerMetrics model and stored locally.
+     */
+    private void updateLocalMetrics() {
+
+        // Initializing the status of components
+        // No need to poll unless node is bootstrapped.
+        Layout layout = serverContext.copyManagementLayout();
+        if (layout == null) {
+            return;
```

Comment:
Please add log message

Reply:
Maybe even throw an exception?
What does it mean if layout equals null?

Reply:
Done.